README.md

```diff
@@ -12,7 +12,6 @@ Kourier is passing the knative serving e2e and conformance tests:
 - [**Getting started**](#getting-started)
 - [**Features**](#features)
 - [**Deployment**](#deployment)
-- [**Development**](#development)
```

Comment:
I guess we should have deleted this with https://github.com/knative-extensions/net-kourier/commit/55bdfaf7570f7e1d78c59880b36f7cddee42f3fa